### PR TITLE
Fix MultipartFormDataBodyParser accept fields

### LINKED ISSUE #2559: Upload image using loopback 4
I'm using loopback 4 latest version. I'm trying to resize and upload my images to local. 

//this is my ImageUploadProvider class.
```ts
import {ImageResizeProvider} from './image-resize.provider';
const path = require('path');
export class ImageUploadProvider {
  constructor() {}
  async uploadImage(req: any) {
    const promise = new Promise(async (resolve, reject) => {
      try {
        let imagePath = path.join(__dirname, '/public/images');
        let fileUpload = new ImageResizeProvider(imagePath);
        if (!req) {
          return resolve({error: 'Please provide an image'});
        }
        const filename = await fileUpload.save(req);
        return resolve({name: filename});
      } catch (error) {
        return reject(error);
      }
    });
  }
}

//this is my ImageResizeProvider class

const sharp = require('sharp');
const uuidv4 = require('uuid/v4');
const path = require('path');

export class ImageResizeProvider {
  private folder: object;
  constructor(folder: object) {
    this.folder = folder;
  }
  async save(buffer: object) {
    const filename = ImageResizeProvider.filename();
    const filepath = this.filepath(filename);
    await sharp(buffer)
      .resize(300, 300, {
        fit: sharp.fit.inside,
        withoutEnlargement: true,
      })
      .toFile(filepath);

    return filename;
  }
  static filename() {
    return `${uuidv4()}.png`;
  }
  filepath(filename: String) {
    return path.resolve(`${this.folder}/${filename}`);
  }
}
```

I need to pass my image in to uploadImage method inside ImageUploadProvider class. 

this is my image upload controller.
```ts
@post('/image/upload', {
    responses: {
      200: {
        content: {
          'application/json': {
            schema: {
              type: 'object',
            },
          },
        },
        description: '',
      },
    },
  })
  async uploadImage(
    @requestBody({
      description: 'multipart/form-data value.',
      required: true,
      content: {
        'multipart/form-data': {
          'x-parser': 'stream',
          schema: {type: 'object'},
        },
      },
    })
    request: Request,
  ): Promise<void> {
    await this.imageUploadProvider.uploadImage("need to pass here");
    console.log(request.image);
  }
```

### PR DESCRIPTION
**Pull request by truongkhanhduy95**
## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [xNew tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/rest/src/__tests__/acceptance/file-upload/file-upload-with-parser.acceptance.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ class MultipartFormDataBodyParser implements BodyParser {
             value: {
               files: request.files,
               // tslint:disable-next-line:no-any
-              fields: (request as any).fields,
+              fields: request.body,
```
Comment (raymondfeng) [inline packages/rest/src/__tests__/acceptance/file-upload/file-upload-with-parser.acceptance.ts]: @truongkhanhduy95 Please elaborate the proposed change? I think `multer` sets `fields`.

**Comment by raymondfeng** [comment]
@truongkhanhduy95 Please amend the commit message per instructions from https://loopback.io/doc/en/lb4/submitting_a_pr.html.

**Comment by raymondfeng** [comment]
@truongkhanhduy95 Are you still interested in moving forward with this PR?

**Comment by truongkhanhduy95** [comment]
Closing this. I have squashed commits by mistake.

**Pull request closed without merging**